src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1

```diff
@@ -1245,6 +1246,222 @@ function Test-PolicyObjectPiping
+    Assert-AreEqual "testExemption" $exemption.Name 
+    Assert-AreEqual Microsoft.Authorization/policyExemptions $exemption.ResourceType
+    Assert-AreEqual "$($rg.ResourceId)/providers/Microsoft.Authorization/policyExemptions/testExemption" $exemption.ResourceId
+    Assert-AreEqual $assignment.ResourceId $exemption.Properties.PolicyAssignmentId
+    Assert-AreEqual "Waiver" $exemption.Properties.ExemptionCategory
+    Assert-AreEqual $description $exemption.Properties.Description
+    Assert-AreEqual $description $exemption.Properties.DisplayName
+    Assert-NotNull $exemption.Properties.Metadata
+    Assert-AreEqual $metadataValue $exemption.Properties.Metadata.$metadataName
+    Assert-Null $exemption.Properties.ExpiresOn
+
+    # get the exemption by name
+    $exemption = Get-AzPolicyExemption -Name testExemption -Scope $rg.ResourceId
+    Assert-AreEqual "testExemption" $exemption.Name 
+    Assert-AreEqual Microsoft.Authorization/policyExemptions $exemption.ResourceType
+    Assert-AreEqual "$($rg.ResourceId)/providers/Microsoft.Authorization/policyExemptions/testExemption" $exemption.ResourceId
+    Assert-AreEqual $assignment.ResourceId $exemption.Properties.PolicyAssignmentId
+    Assert-AreEqual "Waiver" $exemption.Properties.ExemptionCategory
+    Assert-AreEqual $description $exemption.Properties.Description
+    Assert-AreEqual $description $exemption.Properties.DisplayName
+    Assert-NotNull $exemption.Properties.Metadata
+    Assert-AreEqual $metadataValue $exemption.Properties.Metadata.$metadataName
+    Assert-Null $exemption.Properties.ExpiresOn
+
+    # get the exemption by id
+    $exemption = Get-AzPolicyExemption -Id $exemption.ResourceId
+    Assert-AreEqual "testExemption" $exemption.Name 
+    Assert-AreEqual Microsoft.Authorization/policyExemptions $exemption.ResourceType
+    Assert-AreEqual "$($rg.ResourceId)/providers/Microsoft.Authorization/policyExemptions/testExemption" $exemption.ResourceId
+    Assert-AreEqual $assignment.ResourceId $exemption.Properties.PolicyAssignmentId
+    Assert-AreEqual "Waiver" $exemption.Properties.ExemptionCategory
+    Assert-AreEqual $description $exemption.Properties.Description
+    Assert-AreEqual $description $exemption.Properties.DisplayName
+    Assert-NotNull $exemption.Properties.Metadata
+    Assert-AreEqual $metadataValue $exemption.Properties.Metadata.$metadataName
+    Assert-Null $exemption.Properties.ExpiresOn
+
+    # update the policy exemption, validate the result
+    $future1 = [DateTime]::Parse('03/08/3021 23:30:10')
+    $exemption = Set-AzPolicyExemption -Id $exemption.ResourceId -DisplayName testDisplay -ExemptionCategory Mitigated -ExpiresOn $future1 -Metadata '{}'
+    Assert-AreEqual "testDisplay" $exemption.Properties.DisplayName
+    Assert-AreEqual "Mitigated" $exemption.Properties.ExemptionCategory
+    Assert-AreEqual $future1 $exemption.Properties.ExpiresOn
```

Comment:
Updating ExpiresOn  failed in Test-PolicyExemptionCRUD, Test-PolicyExemptionCRUDOnPolicySet and Test-PolicyExemptionCRUDAtManagementGroup, please verify it. Did you record this update test? I can't find 3021-03-08 in recording json files

Reply:
I shouldn't use local time in the test script. Will update it.